earth2studio/data/xr.py

```diff
@@ -62,6 +62,144 @@ def __call__(
         return self.da.sel(time=time, variable=variable)
 
 
+class ModelOutputDatasetSource:
+    """Adapt a model-output ``xarray.Dataset`` into a standardized ``xarray.DataArray``.
+
+    This adapter expects a dataset with an initialization run ``time`` and a
+    forecast ``lead_time`` along with spatial dimensions (``lat``, ``lon``) and
+    one or more data variables. All variables are stacked into a single
+    ``variable`` dimension via ``Dataset.to_array("variable")``. A valid forecast
+    time coordinate is constructed as ``time + lead_time`` and used to replace the
+    original run ``time``; the ``lead_time`` coordinate is then dropped.
+
+    The resulting ``DataArray`` has dimensions in the order
+    ``("time", "variable", "lat", "lon")`` and can be queried by valid forecast
+    time and variable name.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to an xarray-compatible dataset file (e.g., NetCDF/Zarr).
+    filter_dict : dict, optional
+        Dictionary of selections applied before transformation (e.g.,
+        ``{"member": 0}``). Coordinates not in the required
+        dimensions are dropped after selection.
+    **xr_args : Any
+        Additional keyword arguments forwarded to ``xarray.open_dataset``.
+
+    Raises
+    ------
+    ValueError
+        If required input dimensions are missing or if extra dimensions remain
+        after filtering.
+    AssertionError
+        If more than one run initialization ``time`` is present.
+
+    Notes
+    -----
+    - Exactly one initialization ``time`` is required; the valid forecast times
```

Comment:
Switching this to follow the forecast source API would be great to remove this constraint